### PR TITLE
Update partner mock API 🤖

### PR DESCRIPTION
**Pull request by hackebrot**
This pull request is the next one in a series of pull requests adding a working integration test suite (see also #9).

It contains changes to the partner mock API implementation as well as some changes to the `docker-compose.yml` file which will be required for local development. The canned responses from this API are defined in a local file. The path to that file is read from the `RESPONSES_FILE` environment variable. See `partner/app/tests/responses.yml` for an example.

Please let me know if you have any questions!

### CODE REVIEW

**Review by ashrivastava-qa — APPROVED** [review]
LGTM 👍

File: partner/app/main.py [hunk 1]
```diff
@@ -45,20 +45,49 @@ async def read_root():
...
+    # Read response information from the file
+    status_code = response_from_file.status_code
+    content = response_from_file.content
+    delay = response_from_file.delay
+    headers = {header.name: header.value for header in response_from_file.headers}
+
+    if delay:
+        # Add an artificual delay to the handler
+        logger.debug("response is delayed by %s seconds", delay)
+        time.sleep(delay)
```
Comment (pjenvey) [inline partner/app/main.py:75]: This being an async function, you don't want time.sleep blocking the event loop
```suggestion
        asyncio.sleep(delay)
```

File: partner/README.md [hunk 2]
```diff
@@ -10,59 +12,60 @@ Install all requirements via [pip-tools][pip-tools]:
...
+      "id": 12346,
+      "name": "Example",
+      "click_url": "https://example.com/desktop_macos?version=16.0.0",
+      "image_url": "https://example.com/desktop_macos01.jpg",
+      "impression_url": "https://example.com/desktop_macos?id=0001",
+      "advertiser_url": "https://www.example.com/desktop_macos"
+    },
+    {
+      "id": 56790,
+      "name": "Example",
```
Comment (pjenvey) [inline partner/README.md:60]: might want to rename this here and elsewhere to distinguish it from the other, e.g. "Example 2"

**Review by pjenvey — CHANGES_REQUESTED** [review]

File: partner/app/main.py (hunk 1 above)
Comment (hackebrot) [reply partner/app/main.py]: Good catch! I updated this to use `await asyncio.sleep(delay)` instead. We need to `await` to ensure the HTTP request from contile actually times out.

File: partner/README.md (hunk 2 above)
Comment (hackebrot) [reply partner/README.md:60]: I created #17 to update this across all components once we've merged this pull request series.

**Comment by hackebrot** [comment]
Hi @pjenvey! 👋🏻 I addressed the changes you requested and will go ahead and merge this pull request now. Please let me know if you have follow-up questions or comments.

**Pull request merged**